protocols/kad/src/behaviour.rs

```diff
@@ -277,7 +277,7 @@ where
     }
 
     /// Get a mutable reference to internal record store.
-    pub fn store_mut(&mut self) -> &mut TRecordStorage {
+    pub fn store_mut(&mut self) -> &mut RecordStore {
```

Comment:
Why now a trait object, i.e. what's wrong with `&mut TStore`? Also, is it not the case that you then cannot even call `retain` since its not e member of the trait?

Reply:
Oh, you are right, good catch thanks